### PR TITLE
Bug/2257 crash with invalid subscription when matching (release/1.2.0)

### PR DESCRIPTION
**Pull request by kzangeli**
Fixes issue #2257 
Capturing invalid regex for id pattern in subscriptions directly in the parse layer and returning a 400 Bad Request.

### CODE REVIEW

File: CHANGES_NEXT_RELEASE [hunk 1]
```diff
@@ -3,3 +3,4 @@
 - Fix: In some cases, compound attribute values were forwarded as empty strings (Issue #2237)
 - Fix: attribute overwritten when JSON value used in PUT /v2/entities/<eId>/attr/<attrName>/values (Issues #2246 and #2248)
 - Fix: subscription equal filter evaluation on update context logic (#2222)
+- Fix: capturing invalid id patterns at json parse time (Issue #2257)
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE]: I'd suggest to mention in some way that this fix avoids a crash.

File: src/lib/cache/subCache.cpp [hunk 2]
```diff
@@ -72,17 +72,25 @@ volatile SubCacheState subCacheState = ScsIdle;
...
   {
     // FIXME P5: recomp error should be captured? have a look to other usages of regcomp()
     // in order to see how it works
-    regcomp(&entityIdPattern, _entityId.c_str(), 0);
-    entityIdPatternToBeFreed = true;
+    if (regcomp(&entityIdPattern, _entityId.c_str(), 0) != 0)
+    {
+      alarmMgr.badInput(clientIp, "invalid regular expression for idPattern");
+      isPattern = false;  // FIXME P6: this entity should not be let into the system. Must be stopped before
+                          //           right here, best thing todo is simply say it is not a regex
```
Comment (fgalan) [inline src/lib/cache/subCache.cpp]: todo -> to do

Comment (kzangeli) [inline src/lib/cache/subCache.cpp]: Fixed in 9c53592

File: CHANGES_NEXT_RELEASE (hunk 1 above)
Comment (kzangeli) [inline CHANGES_NEXT_RELEASE]: Fixed in 9c53592

**Comment by crbrox** [comment]
LGTM

File: CHANGES_NEXT_RELEASE [hunk 3]
```diff
@@ -3,3 +3,4 @@
 - Fix: In some cases, compound attribute values were forwarded as empty strings (Issue #2237)
 - Fix: attribute overwritten when JSON value used in PUT /v2/entities/<eId>/attr/<attrName>/values (Issues #2246 and #2248)
 - Fix: subscription equal filter evaluation on update context logic (#2222)
+- Fix: capturing invalid id patterns at json parse time - this bug made the broker crash when receiving invalid id-patterns (Issue #2257)
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE]: ...at json subscriptoin parse time...

(I think is useful to delimit the scope of the problem)

Comment (kzangeli) [inline CHANGES_NEXT_RELEASE]: True

Comment (kzangeli) [inline CHANGES_NEXT_RELEASE]: Fixed in f8be773

**Comment by fgalan** [comment]
LGTM

Not deleting branch after merge, to reuse in the PR for develop.

**Pull request merged**